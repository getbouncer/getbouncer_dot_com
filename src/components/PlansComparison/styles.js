/* eslint-disable implicit-arrow-linebreak */
import styled from '@emotion/styled'
import { theme } from '~styles'
import checkedIcon from '~images/checked.svg'

export const Wrapper = styled.section`
  padding: 100px 0;
`

export const Headline = styled.div`
  text-align: center;
  h2 {
    background: ${theme.textGradient};
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    display: inline-block;
    font-size: 4rem;
    line-height: 1.1;
    margin-bottom: 50px;
  }
`

export const Grid = styled.div`
  display: block;
`

export const Row = styled.div`
  display: grid;
  grid-template-columns: 3fr 2fr 2fr 2fr;

  &:first-of-type {
    border-bottom: 2px solid ${theme.colors.blue};

    > div {
      display: block;
      text-align: center;
      padding-bottom: 20px;
    }
  }

  &:not(:first-of-type) {
    > div {
      align-items: center;
      height: 65px;

      &:not(:first-of-type) {
        justify-content: center;
      }
    }

    &:not(:last-of-type) {
      border-bottom: 1px solid ${theme.colors.lightBlue};
    }
  }
`

export const Cell = styled.div`
  display: flex;

  &:not(:last-child) {
    border-right: 1px solid ${theme.colors.lightBlue};
  }

  ${({ checked }) =>
    checked &&
    `
      position: relative;

      &::after {
        content: '';
        background-image: url(${checkedIcon});
        background-position: 50%;
        background-repeat: no-repeat;
        background-size: 12px;
        background-color: ${theme.colors.blue};
        border-radius: 100%;
        position: absolute;
        height: 22px;
        width: 22px;
      }
    `}
`
