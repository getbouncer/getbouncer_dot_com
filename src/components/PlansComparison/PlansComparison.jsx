import React from 'react'
import PropTypes from 'prop-types'
import Container from '~globals/Container'
import { Cell, Grid, Headline, Row, Wrapper } from './styles'

const PricingPlans = ({ data }) => (
  <Wrapper>
    <Container>
      <Headline>
        <h2>{data.comparison_headline.text}</h2>
      </Headline>
      <Grid>
        <Row>
          <Cell />
          {data.body.map(cell => (
            <Cell key={cell.id}>
              <h3>{cell.primary.tier_name}</h3>
              <span>{cell.primary.tier_short_description}</span>
            </Cell>
          ))}
        </Row>
        <Row>
          <Cell>Scan any card</Cell>
          <Cell checked />
          <Cell checked />
          <Cell checked />
        </Row>
        <Row>
          <Cell>Verify card in real time</Cell>
          <Cell />
          <Cell checked />
          <Cell checked />
        </Row>
        <Row>
          <Cell>24/7 support, email, and live chat</Cell>
          <Cell checked />
          <Cell checked />
          <Cell checked />
        </Row>
        <Row>
          <Cell>SLA guarantees</Cell>
          <Cell />
          <Cell checked />
          <Cell checked />
        </Row>
        <Row>
          <Cell>Custom UI</Cell>
          <Cell checked />
          <Cell checked />
          <Cell checked />
        </Row>
        <Row>
          <Cell>Live integration help</Cell>
          <Cell />
          <Cell />
          <Cell checked />
        </Row>
        <Row>
          <Cell>Bouncerâ€™s Fraud Score</Cell>
          <Cell />
          <Cell />
          <Cell checked />
        </Row>
      </Grid>
    </Container>
  </Wrapper>
)

PricingPlans.propTypes = {
  data: PropTypes.shape().isRequired,
}

export default PricingPlans
