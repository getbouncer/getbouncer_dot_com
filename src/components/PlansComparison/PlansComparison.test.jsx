import React from 'react'
import renderer from 'react-test-renderer'
import { describe, expect, it } from 'jest'
import PlansComparison from './PlansComparison'

describe('PlansComparison', () => {
  it('renders correctly', () => {
    const data = {
      headline: {
        text: 'headline',
      },
      comparison_headline: {
        text: 'comparison headline',
      },
      text: {
        text: 'text',
      },
      body: [
        {
          id: 'id 1',
          primary: {
            tier_link_internal: 'https://www.example.com/tier_1',
            tier_link: 'https://www.example.com/tier_1',
            tier_cta: 'tier 1 cta',
            tier_description: 'tier 1 description',
            tier_name: 'tier 1',
            tier_short_description: 'tier 1 short description',
            tier_price: 'tier 1 price',
          },
          items: [
            {
              tier_feature: 'tier 1 feature',
            },
            {
              tier_feature: 'tier 1 feature',
            },
          ],
        },
        {
          id: 'id 2',
          primary: {
            tier_link_internal: 'https://www.example.com/tier_2',
            tier_link: 'https://www.example.com/tier_2',
            tier_cta: 'tier 2 cta',
            tier_description: 'tier 2 description',
            tier_name: 'tier 2',
            tier_short_description: 'tier 2 short description',
            tier_price: 'tier 2 price',
          },
          items: [
            {
              tier_feature: 'tier 2 feature',
            },
            {
              tier_feature: 'tier 2 feature',
            },
          ],
        },
      ],
    }
    const tree = renderer
      .create(<PlansComparison data={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
