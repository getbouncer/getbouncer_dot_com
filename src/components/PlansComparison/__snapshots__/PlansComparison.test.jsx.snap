// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlansComparison renders correctly 1`] = `
<section
  className="css-my4pmd"
>
  <div
    className="css-1fywgw4"
  >
    <div
      className="css-kxswky"
    >
      <h2>
        comparison headline
      </h2>
    </div>
    <div
      className="css-1i82ydo"
    >
      <div
        className="css-1n9sbj9"
      >
        <div
          className="css-tdepjq"
        />
        <div
          className="css-tdepjq"
        >
          <h3>
            tier 1
          </h3>
          <span>
            tier 1 short description
          </span>
        </div>
        <div
          className="css-tdepjq"
        >
          <h3>
            tier 2
          </h3>
          <span>
            tier 2 short description
          </span>
        </div>
      </div>
      <div
        className="css-1n9sbj9"
      >
        <div
          className="css-tdepjq"
        >
          Scan any card
        </div>
        <div
          checked={true}
          className="css-sma4tl"
        />
        <div
          checked={true}
          className="css-sma4tl"
        />
        <div
          checked={true}
          className="css-sma4tl"
        />
      </div>
      <div
        className="css-1n9sbj9"
      >
        <div
          className="css-tdepjq"
        >
          Verify card in real time
        </div>
        <div
          className="css-tdepjq"
        />
        <div
          checked={true}
          className="css-sma4tl"
        />
        <div
          checked={true}
          className="css-sma4tl"
        />
      </div>
      <div
        className="css-1n9sbj9"
      >
        <div
          className="css-tdepjq"
        >
          24/7 support, email, and live chat
        </div>
        <div
          checked={true}
          className="css-sma4tl"
        />
        <div
          checked={true}
          className="css-sma4tl"
        />
        <div
          checked={true}
          className="css-sma4tl"
        />
      </div>
      <div
        className="css-1n9sbj9"
      >
        <div
          className="css-tdepjq"
        >
          SLA guarantees
        </div>
        <div
          className="css-tdepjq"
        />
        <div
          checked={true}
          className="css-sma4tl"
        />
        <div
          checked={true}
          className="css-sma4tl"
        />
      </div>
      <div
        className="css-1n9sbj9"
      >
        <div
          className="css-tdepjq"
        >
          Custom UI
        </div>
        <div
          checked={true}
          className="css-sma4tl"
        />
        <div
          checked={true}
          className="css-sma4tl"
        />
        <div
          checked={true}
          className="css-sma4tl"
        />
      </div>
      <div
        className="css-1n9sbj9"
      >
        <div
          className="css-tdepjq"
        >
          Live integration help
        </div>
        <div
          className="css-tdepjq"
        />
        <div
          className="css-tdepjq"
        />
        <div
          checked={true}
          className="css-sma4tl"
        />
      </div>
      <div
        className="css-1n9sbj9"
      >
        <div
          className="css-tdepjq"
        >
          Bouncerâ€™s Fraud Score
        </div>
        <div
          className="css-tdepjq"
        />
        <div
          className="css-tdepjq"
        />
        <div
          checked={true}
          className="css-sma4tl"
        />
      </div>
    </div>
  </div>
</section>
`;
